src/systems/filecoin_blockchain/chainsync/_index.md

```diff
@@ -345,22 +347,23 @@ State Machine:
 - Validation computation is considerable, and a serious DOS attack vector.
 - Secure implementations must carefully schedule validation and minimize the work done by pruning blocks without validating them fully.
 - `ChainSync` SHOULD keep a cache of unvalidated blocks (ideally sorted by likelihood of belonging to the chain), and delete unvalidated blocks when they are passed by `FinalityTipset`, or when `ChainSync` is under significant resource load.
-- It is key to note that any block received after the `ROUND_CUTOFF` time must be automatically discarded by the miner until the start of the next epoch.
+- It is key to note that any block received after the `EPOCH_CUTOFF` time must be automatically discarded by the miner until the start of the next epoch.
 
 - **Progressive Stages of Block Validation**
   - _(TODO: move this to blockchain/Block section)_
   - **BV0 - Syntactic Validation**: Validate data structure packing and ensure correct typing.
-  - **BV1 - Light Consensus State Checks**: Validate `b.ChainWeight`, `b.ChainEpoch`, `b.MinerAddress`, `b.Timestamp`, are plausible (some ranges of bad values can be detected easily, especially if we have the state of the chain at `b.ChainEpoch - consensus.LookbackParameter`. Eg Weight and Epoch have well defined valid ranges, and `b.MinerAddress`
-  must exist in the lookback state). This requires some chain state, enough to establish plausibility levels of each of these values. A node should be able to estimate valid ranges for `b.ChainEpoch` based on the `LastTrustedCheckpoint`. `b.ChainWeight` is easy if some of the relatively recent chain is available, otherwise hard.
-  - **BV2 - Signature Validation**: Verify `b.BlockSig` is correct.
-  - **BV3 - Verify ElectionProof**: Verify `b.ElectionProof` is correct. This requires having state for relevant lookback parameters.
-  - **BV4 - Verify Ancestry links to chain**: Verify ancestry links back to trusted blocks. If the ancestry forks off before finality, or does not connect at all, it is a bad block.
-  - **BV4 - Verify MessageSigs**: Verify the signatures on messages
-  - **BV5 - Verify StateTree**: Verify the application of `b.Parents.Messages()` correctly produces `b.StateTree` and `b.MessageReceipts`
+  - **BV1 - Epoch Window Validation**: Validate that block was produced during the current epoch.  Validate that the block was received before `EPOCH_CUTOFF`.  This stage only applies during `CHAIN_FOLLOW`.
```

Comment:
Note to future spec writers: as we may continue to split up and add to these the current number system and presentation format could use some rethinking.

Reply:
I feel like BV1 may be more subtle (assuming I got it):
- discard any block from future or from past validation ago
- cache other blocks (in case returning to CATCH UP mode) but do not consider them for inclusion in head

Reply:
OK I'm going to go back to work on this one.

Thinking this over more with help from @jzimmerman the right place to encode this information is in the chain selection function (set the chain to head only if its height is our current epoch).  We really want to avoid adding more complexity (what is this cache?  how does it work? how is it used?) directly to the chain sync system.

Reply:
I would agree. The cache I mean is you should keep blocks around to not request them from the network if you get a block much heavier than yours in a given epoch from a subchain you didn't know about. You probably already have part of the chain so you want to request as few as possible when going back to catch up. This is what Juan meant I believe when saying "only discard past finality".

So agree all in all. the effect of your call is just that chain head will only be set to head if validated to current epoch. Else youll fall behind enough to move back to catch up and sync to new best chain